### DISCUSSION TITLE
How to restart the software developed by fyne

### DISCUSSION THREAD

**Discussion by junmaqiang** [fyne-io/fyne#3721] [Q&A] [answered]
Any friends to provide a reference method?

**Comment by Jacalz** [comment]
I am not entirely sure what you are asking for but if it is to make the currently open application restart, then I believe that you likely will find your answer in this reply to another discussion: https://github.com/fyne-io/fyne/discussions/3538#discussioncomment-4585531. In short this is the answer:

> I think asking the application to restart itself is going to be quite hard to get working across platforms; or will at least require platform specific solutions for most of them. I suppose https://stackoverflow.com/questions/41376092/golang-how-to-auto-restart-process-when-binary-updated could be a useful place to start. 

In general, it is best to avoid restarting if you can. There might be a better solution depending on what you are trying to do.

**Comment by andydotxyz** [comment, marked as answer]
I can confirm that this is a very complex topic and @Jacalz is right it is not a good thing to do.
The only use-case I know of is if you have to upgrade the software - which is what the Fyne Labs selfupdate does, so you could look there for clues or even call the `Restart` function directly. https://github.com/fynelabs/selfupdate

**Reply by junmaqiang** [reply]
thank you very much
